### PR TITLE
HDR block

### PR DESCRIPTION
**Pull request by rgclayton**
New HDR block to support HDR images

* Display HDR images
* Supports before and after images that swap on hover.

Resolves: [MWPW-136514](https://jira.corp.adobe.com/browse/MWPW-136514)

**Test URLs:**
* Before: https://main--blog--adobecom.hlx.page/en/drafts/rclayton/hdr-explained?martech=off
* After: https://hdr--blog--adobecom.hlx.page/en/drafts/rclayton/hdr-explained?martech=off

### CODE REVIEW

File: blocks/hdr/hdr.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.hdr {
+  margin: 40px 0;
+}
+.hdr .caption {
```
Comment (chrischrischris) [inline blocks/hdr/hdr.css:4]: supa nit: blank line

File: blocks/hdr/hdr.css [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  left: 0;
+}
+
+.hdr .hdr-container.multiple:hover img:last-child {
+  display: block;
+  z-index: 2;
+}
+
+.hdr .hdr-container.multiple:hover:after {
+  content: 'HDR';
```
Comment (chrischrischris) [inline blocks/hdr/hdr.css:31]: I assume this won't need to be localized?

File: blocks/hdr/hdr.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export default async function init(el) {
+  const links = el.querySelectorAll(':scope a');
+  if (!links.length) return;
+
+  const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+  const caption = el.querySelector(':scope em');
+  const container = createTag('div', { class: 'hdr-container' }, null);
+  el.innerHTML = '';
+
+  if(links.length > 1) {
```
Comment (chrischrischris) [inline blocks/hdr/hdr.js:22]: nit: `if (` (also line 36)

File: blocks/hdr/hdr.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const mediaUrl = new URL(a.href);
+    return mediaUrl.pathname;
+  } catch (e) {
+    console.log('Error parsing media url', e);
+  }
+  return '';
+};
+
+export default async function init(el) {
+  const links = el.querySelectorAll(':scope a');
```
Comment (chrischrischris) [inline blocks/hdr/hdr.js:14]: I assume if the block is being imported then it should always have at least one link?

If that's the case I think it's ok to import utils at top of file.

**Review by chrischrischris — APPROVED** [review]

File: blocks/hdr/hdr.css (hunk 2 above)
Comment (rgclayton) [reply blocks/hdr/hdr.css:31]: Good point. My assumption is no. But, I'll have to double check.

File: blocks/hdr/hdr.js (hunk 4 above)
Comment (rgclayton) [reply blocks/hdr/hdr.js:14]: True.

Comment (auniverseaway) [inline blocks/hdr/hdr.js:14]: If you're not doing a direct descendant, you don't need `:scope`.

Gold plating:

```js
querySelectorAll('a[href*=".jpg"], a[href*=".png"]'); // image formats you want to support.
```

File: blocks/hdr/hdr.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  const container = createTag('div', { class: 'hdr-container' }, null);
+  el.innerHTML = '';
+
+  if(links.length > 1) {
+    container.classList.add('multiple');
+  }
+  el.append(container);
+
+  links.forEach((link) => {
+    const img = document.createElement('img');
```
Comment (auniverseaway) [inline blocks/hdr/hdr.js:28]: I would either use createTag or createElement, not both.

File: blocks/hdr/hdr.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    img.src = getMediaFilename(link);
+    img.setAttribute('loading', 'lazy');
+    container.append(img);
+    link.remove();
+  });
+
+  if(!caption) return;
+  const para = document.createElement('p');
```
Comment (auniverseaway) [inline blocks/hdr/hdr.js:37]: I would either use createTag or createElement, not both.

**Review by auniverseaway — APPROVED** [review]
Small suggestions, nothing critical.

File: blocks/hdr/hdr.js (hunk 6 above)
Comment (rgclayton) [reply blocks/hdr/hdr.js:37]: Good catch! I did a small refactor to use `createTag` and missed this one.

**Comment by rgclayton** [comment]
Updated based on feedback. Thank you for looking this over!

**Pull request merged**